### PR TITLE
Confirm button should be disabled if account has no balance

### PR DESCRIPTION
**Pull request by jpuri**
Fixes regression issue about confirm button being enabled even if account has no balance.

Fix is small its the first commit. More changes were needed to write test cases 2,3 commit are done for test cases.

### CODE REVIEW

File: app/components/UI/ApproveTransactionHeader/ApproveTransactionHeader.tsx [hunk 1]
```diff
@@ -152,9 +146,9 @@ const ApproveTransactionHeader = ({
...
         accountAddress={activeAddress}
-        accountNativeCurrency={accountInfo.currency}
-        accountBalance={accountInfo.balance}
-        accountName={accountInfo.accountName}
+        accountNativeCurrency={accountCurrency}
+        accountBalance={accountBalance}
+        accountName={accountName}
         accountBalanceLabel={strings('transaction.balance')}
         accountNetwork={networkName}
         badgeProps={{
```
Comment (jpuri) [inline app/components/UI/ApproveTransactionHeader/ApproveTransactionHeader.tsx:154]: This class using combination of object in state which are dependency for `useMemo` was creating issues in writing test cases. Thus I refactored it.

File: app/components/UI/TransactionReview/index.test.tsx [hunk 2]
```diff
@@ -3,24 +3,53 @@ import TransactionReview from './';
 import configureMockStore from 'redux-mock-store';
 import { shallow } from 'enzyme';
 import { Provider } from 'react-redux';
+import renderWithProvider from '../../../util/test/renderWithProvider';
+import Engine from '../../../core/Engine';
 
-const generateTransform = jest.fn();
-const mockStore = configureMockStore();
-const initialState = {
+jest.mock('react-native-keyboard-aware-scroll-view', () => {
```
Comment (segun) [inline app/components/UI/TransactionReview/index.test.tsx:9]: Hi @jpuri can you please explain why we need to mock this. Thanks.

Comment (jpuri) [reply app/components/UI/TransactionReview/index.test.tsx:9]: Hey I do not remember exact issue, but it gives exception in running rect-native test cases.

File: app/components/UI/TransactionReview/index.test.tsx [hunk 3]
```diff
@@ -41,28 +69,97 @@ const initialState = {
...
+  it('should not have confirm button disabled if from account has balance', () => {
+    const { getByRole } = renderWithProvider(
+      <TransactionReview
+        EIP1559GasData={{}}
+        generateTransform={generateTransform}
+      />,
+      { state: mockState },
+    );
+    const confirmButton = getByRole('button', { name: 'Confirm' });
+    expect(confirmButton.props.disabled).toBe(true);
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/components/UI/TransactionReview/index.test.tsx]: Good point @digiwand , I updated the PR with code change.

File: app/components/UI/TransactionReview/index.test.tsx (hunk 2 above)
Comment (jpuri) [reply app/components/UI/TransactionReview/index.test.tsx:11]: This is the error
```
    TypeError: Cannot read properties of undefined (reading 'major')

      at Object.extractNativeRef (node_modules/react-native-keyboard-aware-scroll-view/lib/KeyboardAwareHOC.js:135:116)
      at _this._handleRef (node_modules/react-native-keyboard-aware-scroll-view/lib/KeyboardAwareHOC.js:508:55)
      at commitAttachRef (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:11992:7)
      at commitLayoutEffects (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:14448:9)
      at Object.invokeGuardedCallbackProd (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:11308:10)
      at invokeGuardedCallback (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:11499:31)
      at commitRootImpl (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:14172:9)
      at unstable_runWithPriority (node_modules/react-test-renderer/node_modules/scheduler/cjs/scheduler.development.js:468:12)
```

**Review by blackdevelopa — APPROVED** [review]
Outside of the current reviews. LGTM

**Comment by chrisleewilcox** [comment]
@jpuri @blackdevelopa can we rebase this PR off of branch `release/6.5.0`.  This needs to go straight into the release.

**Review by digiwand — APPROVED** [review]

**Comment by seaona** [comment]
The fix looks good. I've opened 2 other issues related to that, but can be fixed in separate PRs:
- [[Bug] Next button not disabled when there is not enough balance to proceed](https://github.com/MetaMask/mobile-planning/issues/935#top)
- [[Bug] Warning for not enough funds only appears when we go to Edit fees and back to the Confirmation screen](https://github.com/MetaMask/mobile-planning/issues/937#top)

**Pull request merged**